### PR TITLE
Move pdns-sysrepo configuration to sysrepo

### PR DESCRIPTION
**Pull request by pieterlexis**
This also updates the dependencies to released versions of libyang and sysrepo.

### CODE REVIEW

File: src/config/config.cc [hunk 1]
```diff
@@ -42,20 +51,62 @@ int Config::module_change(sysrepo::S_Session session, const char* module_name,
...
+    try {
+      auto changeIter = session->get_changes_iter("/pdns-server:pdns-sysrepo//*");
+      auto change = session->get_change_tree_next(changeIter);
+      while (change) {
+        if (change->node()->schema()->nodetype() == LYS_LEAF) {
+          auto leaf = std::make_shared<libyang::Data_Node_Leaf_List>(change->node());
+          spdlog::trace("pdns-sysrepo config change path={} value={}", leaf->path(), leaf->value_str());
+          if (change->node()->path() == "/pdns-server:pdns-sysrepo/pdns-service/config-file") {
+            d_pdns_conf = leaf->value_str();
+          }
```
Comment (Habbie) [inline src/config/config.cc]: would `else if` be clearer here and a few more times?

File: src/config/config.cc [hunk 2]
```diff
@@ -42,20 +51,62 @@ int Config::module_change(sysrepo::S_Session session, const char* module_name,
...
+
+  if (event == SR_EV_DONE) {
+    if (d_skip_done) {
+      d_skip_done = false;
+      return SR_ERR_OK;
+    }
+    try {
+      auto changeIter = session->get_changes_iter("/pdns-server:pdns-sysrepo//*");
+      auto change = session->get_change_tree_next(changeIter);
+      while (change) {
```
Comment (Habbie) [inline src/config/config.cc]: `while (auto change = session->get....)` ?

File: src/configurator/subscribe.cc [hunk 3]
```diff
@@ -89,11 +89,11 @@ int ServerConfigCB::module_change(sysrepo::S_Session session, const char* module
...
 
       try {
-        spdlog::debug("Moving {} to {}", fpath, privData["fpath"]);
-        fs::rename(fpath, privData["fpath"]);
+        spdlog::debug("Moving {} to {}", fpath, d_config->getPdnsConfigFilename());
+        fs::rename(fpath, d_config->getPdnsConfigFilename());
       }
       catch (const exception& e) {
-        spdlog::warn("Unable to move {} to {}: {}", fpath, privData["fpath"], e.what());
+        spdlog::warn("Unable to move {} to {}: {}", fpath, d_config->getPdnsConfigFilename(), e.what());
```
Comment (Habbie) [inline src/configurator/subscribe.cc]: remove the tempfile in this case?

File: documentation/guides/config.rst [hunk 4]
```diff
@@ -1,12 +1,33 @@
...
-By default, this file is ``/etc/pdns-sysrepo/pdns-sysrepo.yaml``, but this can be changed using the ``-c`` command line argument.
+This software is configured with sysrepo under the ``/pdns-server:pdns-sysrepo/`` tree.
+As this is in the YANG datastore, these settings can be changed at runtime.
 
-These are the default settings and their descriptions:
+This tree has 2 containers, ``pdns-service`` and ``logging``.
 
-.. literalinclude:: ../../pdns-sysrepo.example.yaml
-   :language: yaml
+Configuring logging
```
Comment (Habbie) [inline documentation/guides/config.rst]: was the include removed on purpose?

Comment (pieterlexis) [reply documentation/guides/config.rst]: yes, as there will be no more YAML config after this PR :)

Comment (Habbie) [reply documentation/guides/config.rst]: duh!

**Pull request merged**